lib/autoload/kuroko2/workflow/engine.rb

```diff
@@ -164,6 +185,15 @@ def notify_long_elapsed_time_if_needed(token)
           Kuroko2.logger.info(message)
         end
       end
+
+      def sleep_for_each_retry(node, token)
+        if token.context['RETRY'].present? && token.context['RETRY'][node.path].present?
```

Comment:
do you really want to validate althogh this method is called after line 103-105?